ksso/main.py

```diff
@@ -0,0 +1,143 @@
+│ This is a small CLI application that helps authenticate keycloak         │
+│ users into AWS via OIDC.                                                 │
+│                                                                          │
+│ We obtain a token from keycloak, then we obtain short-term IAM           │
+│ credential by assuming role with web-indentity token.                    │
+│                                                                          │
+│ The AWS role to assume should contain proper trust policy for the        │
+│ identity provider associated with the keycloak instance.                 │
+│                                                                          │
+└──────────────────────────────────────────────────────────────────────────┘
+"""
+
+import argparse
+import logging
+import os
+import threading
+import webbrowser
+from queue import Queue
+from sys import exit
+
+import jwt
+import requests
+
+from ksso.aws import (
+    assume_aws_role_with_keycloak_token,
+    export_aws_credentials_env,
+    export_aws_credentials_json,
+)
+from ksso.config import ConfigError, load_config
+from ksso.server import app
+
+DEFAULT_CONFIG_PATH = os.path.expanduser("~/.ksso_config.toml")
+
+# Thread-safe queue for inter-thread communication as we need to pass
+# from the web-server thread back into the main process
+token_queue = Queue()
+app.token_queue = token_queue
+logger = logging.getLogger("error")
+
+
+def main():
+    """
+    Entry point for the application.
+
+    This function:
+    1. Parses CLI arguments for configuration and options.
+    2. Sets up the Flask server for handling Keycloak's OIDC authentication
+       callback.
+    3. Opens the browser to initiate the Keycloak login flow.
+    4. Waits for authentication to complete and retrieves the access token.
+    5. Uses the access token to assume an AWS IAM role and outputs credentials.
+    """
+    parser = argparse.ArgumentParser(
+        description="Authenticate via Keycloak and assume an AWS role."
+    )
+    parser.add_argument(
+        "--json", action="store_true", help="Output AWS credentials in JSON format."
+    )
+    parser.add_argument(
+        "--env",
+        action="store_true",
+        help="Output AWS credentials in bash source format.",
+    )
+    parser.add_argument(
+        "--config",
+        type=str,
+        default=DEFAULT_CONFIG_PATH,
+        help=f"Path to the TOML config file (default: {DEFAULT_CONFIG_PATH})",
+    )
+    parser.add_argument(
+        "--client-id",
+        type=str,
+        required=True,
+        help=(
+            "Keycloak client id (typically after the name of the client), "
+            "where <name> is the actual client name/prefix"
+        ),
+    )
+    parser.add_argument(
+        "--aws-role-arn",
+        type=str,
+        required=True,
+        help="AWS IAM Role ARN to assume (e.g., arn:aws:iam::123456789012:role/role-name)",
+    )
+    args = parser.parse_args()
+
+    try:
```

Comment:
I would move this `keycloack` auth to the `config.py` file too, so all logic related to keycloack config would be there and this `main` function would be easier to read and understand

Reply:
Done, pls check